### DISCUSSION TITLE
How to access the firebase firestore in this module?

### DISCUSSION THREAD

**Discussion by SelfDevTV** [gladly-team/next-firebase-auth#110] [Q&A]
Hi guys,

I want to use this package for my next web app. 

2 Questions:

* How and where to access the firestore and other things from firebase? For example when a user signs up, I want to store it in the database, mabye add some attributes to it. Where and how to do this?
* Any plans to build it with react hooks / context? So we don't have to use a HOC?

Thx guys the package looks really interesting!

**Comment by SelfDevTV** [comment]
I think I figured out my first question. Just to make sure. 

Is this the correct usage of this package? I started with the getting started guide and then added the login and logout button:

```
import React, { useEffect } from "react";
import {
  useAuthUser,
  withAuthUser,
  withAuthUserTokenSSR,
} from "next-firebase-auth";
import firebase from "firebase";

const Demo = () => {
  const AuthUser = useAuthUser();

  useEffect(() => {
    console.log(AuthUser);
  });

  const login = () => {
    var provider = new firebase.auth.FacebookAuthProvider();
    firebase
      .auth()
      .signInWithPopup(provider)
      .then((result) => {
        console.log(result);
        firebase.firestore().collection("users").doc(result.user.email).set({
          email: result.user.email,
          displayName: result.user.displayName,
        });
      });
  };

  const logout = () => {
    firebase.auth().signOut();
  };

  return (
    <div>
      <p>Your email is {AuthUser.email ? AuthUser.email : "unknown"}.</p>

      <button onClick={() => login()}>Login</button>
      <button onClick={() => logout()}>Logout</button>
    </div>
  );
};

// Note that this is a higher-order function.
export const getServerSideProps = withAuthUserTokenSSR()();

export default withAuthUser()(Demo);
```

**Reply by kmjennison** [reply]
Yes. Conversation continued at #113.

**Comment by kmjennison** [comment]
Regarding hooks: could you recommend how you'd implement this? Definitely open to other approaches. We chose a HOC pattern because, in conjunction with `getServerSideProps`, it removes the additional props created by this package.
